pkg/machine/qemu/machine.go

```diff
@@ -1287,23 +1334,74 @@ func (v *MachineVM) waitAPIAndPrintInfo(forwardState apiForwardingState, forward
 	}
 }
 
-func (v *MachineVM) writeConfig() error {
-	// GetConfDir creates the directory so no need to check for
-	// its existence
-	vmConfigDir, err := machine.GetConfDir(vmtype)
+// update returns the content of the VM's
+// configuration file in json
+func (v *MachineVM) update() error {
+	if err := v.setConfigPath(); err != nil {
+		return err
+	}
+	b, err := v.ConfigPath.Read()
 	if err != nil {
 		return err
 	}
+	if os.IsNotExist(err) {
+		return errors.Wrap(machine.ErrNoSuchVM, v.Name)
+	}
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(b, v)
+	if err != nil {
+		migrateErr := migrateVM(v.ConfigPath.GetPath(), b, v)
```

Comment:
looks to me that there is no reason to not just use err =  migrateVM(v.ConfigPath.GetPath(), b, v) here.

You always set the err back to nil if you don't get an error.